Scripts/AnimationScripts/PlayMyAnimation.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayMyAnimation : MonoBehaviour
+{
+    [SerializeField]
+    private Animator myAnimationController;
+
+    public static event System.Action OnDoorOpened;
+
+    public enum MySign
+    {
+        Left,
+        Right
+    }
+
+    public MySign currentMysign;
+
+    IEnumerator LeftOpenDoorAnim()
+    {
+        Vector3 euler = transform.localEulerAngles;
+        euler.z += 90;
+        transform.localEulerAngles = euler;
+        float timer = 0f;
+
+        while(true)
+        {
+            timer += Time.deltaTime*.25f;
+            transform.localRotation = Quaternion.Lerp(transform.localRotation,Quaternion.Euler(euler),timer);
+
+            if (timer >= 1)
+            {
+                break;
+            }
+
+            yield return new WaitForEndOfFrame();
+        }
+    }
+
+    IEnumerator RightOpenDoorAnim()
+    {
+        Vector3 euler = transform.localEulerAngles;
+        euler.z -= 90;
+        transform.localEulerAngles = euler;
+        float timer = 0f;
+
+        while (true)
+        {
+            timer += Time.deltaTime * .25f;
+            transform.localRotation = Quaternion.Lerp(transform.localRotation, Quaternion.Euler(euler), timer);
+
+            if (timer >= 1)
+            {
+                break;
+            }
+
+            yield return new WaitForEndOfFrame();
+        }
+    }
```

Comment:
Sadece sağa veya sola dönme açısı değişiyor, tek methodla yazmak çok daha kolay ve anlaşılır olacaktır.

```suggestion

    IEnumerator OpenDoorAnim(Direction direction)
    {
        Vector3 euler = transform.localEulerAngles;
        euler.z += (direction == Direction.Left) ? 90 : -90;
        transform.localEulerAngles = euler;
        float timer = 0f;

        while(true)
        {
            timer += Time.deltaTim *.25f;
            transform.localRotation = Quaternion.Lerp(transform.localRotation,Quaternion.Euler(euler),timer);

            if (timer >= 1)
            {
                break;
            }

            yield return new WaitForEndOfFrame();
        }
    }
```